tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/DdwafObjectStruct.cs

```diff
@@ -42,10 +42,50 @@ internal struct DdwafObjectStruct
         [FieldOffset(32)]
         public DDWAF_OBJ_TYPE Type;
 
-        internal IReadOnlyDictionary<string, string[]> Decode()
+        internal object Decode()
+        {
+            object res = Type switch
+            {
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_STRING => Marshal.PtrToStringAnsi(Array, (int)NbEntries),
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_SIGNED => IntValue,
```

Comment:
Maybe we could avoid boxing here by testing on the type before and having a function upfront for value types specifically if its `DDWAF_OBJ_SIGNED`, also we dont need to process `DDWAF_OBJ_UNSIGNED` ? 🤔

Reply:
Although they are not used, I've added Unsigned also for completeness. Boxing should be minimal (or occur at all) because value types are not returned currently by the Waf.